### PR TITLE
Replacing list appending based input into model.fit() with dictionary based input and use default loss function

### PR DESCRIPTION
**Pull request by theo1701**
The combination of the two updates remove the need to append `w` to the end of the `y`, making it more efficient and easier to read.
The default loss function has been running normally thus far. I still need to observe its behaviour a bit more on mc data.
Distributed training had been tested and removed. I think it will try to create a dataset from numpy first and then apply auto-sharding. The effect is much worse runtime that is almost the level of the tfdata branch without any optimizations.

### CODE REVIEW

**Comment by theo1701** [comment]
The number of parallel models in parallel is also made an argument to make (automated) testing easier.

**Comment by theo1701** [comment]
Merged recent changes to master into this branch. During testing there was a one time error that I am unable to replicate. I suspect that it was because ml2's GPU was being used.

File: python/modelUtils.py [hunk 1]
```diff
@@ -174,25 +174,29 @@ def get_model(input_shape, nclass=2, model_name='dense_100x3'):
...
+
+    train_w, val_w = [], []
+
+    X_train, X_val, Y_train, Y_val = train_test_split(X, Y, random_state=325)
 
-    # prepare the lists
+    # prepare the dictionaries
     for i in range(n_models_in_parallel):
-        X_train, X_val, Y_train, Y_val, w_train, w_val = train_test_split(X, Y, w[i])
+        w_train, w_val = train_test_split(w[i], random_state=325)
```
Comment (zctao) [inline python/modelUtils.py:187]: The purpose of using an integer for the random_state is to ensure the train test splitting is the same for data arrays and for all weight arrays right? Instead of always using a hard coded magic number 325, would it be possible to get a random number every time train model is called and use it as the random_state?

Comment (theo1701) [reply python/modelUtils.py]: Done. Replaced it with a dynamically generated random state that changes each time train_model is called but the same across each parallel models.
